riscv/src/compiler.rs

```diff
@@ -1346,6 +1431,54 @@ fn process_instruction(instr: &str, args: &[Argument]) -> Vec<String> {
             only_if_no_write_to_zero(format!("{rd} <== load_label({label});"), rd)
         }
 
+        // atomic and synchronization
```

Comment:
```suggestion
        // atomic and synchronization. We implement a single-threaded machine, so this is a no-op
```

Reply:
This is like, the section name, like we have `// memory access` and `// jump and call`. Not every atomic instruction is a nop.